Sources/Common/Communication/EventBus.swift

```diff
@@ -1,140 +1,104 @@
-        self.listenersRegistry = listenersRegistry
+    deinit {
+        removeAllObservers()
     }
 
-    /// Sends an event to the corresponding listeners.
-    ///
-    /// - Parameter event: The event to be sent. Must conform to `EventRepresentable`.
-    @discardableResult public func send<E>(_ event: E) -> Bool where E: EventRepresentable {
-        guard let listener = listenersRegistry.getListener(forEventType: E.self) else {
-            return false
-        }
-        listener.send(event)
-        return true
+    init() {
+        DIGraphShared.shared.logger.debug("SharedEventBus initialized")
     }
 
-    /// Registers a listener for a specific event type and performs the provided action when that event is emitted.
-    ///
+    /// Posts an event to the EventBus.
     /// - Parameters:
-    ///   - eventType: The type of event to listen for.
-    ///   - action: The action to perform when an event of the specified type is emitted.
-    /// - Returns: A cancellable instance used to unsubscribe from the event.
+    ///   - event: The event to be posted.
+    ///   - queue: An optional DispatchQueue on which to post the event. If nil, uses the current queue.
+    /// - Returns: Boolean indicating if there were any observers for the event.
     @discardableResult
-    public func onReceive<E: EventRepresentable>(_ eventType: E.Type, perform action: @escaping (E) -> Void) -> AnyCancellable {
-        subscribeAndNotify(eventType: eventType, action: action)
-    }
+    public func post<E>(_ event: E, on queue: DispatchQueue? = nil) -> Bool where E: EventRepresentable {
+        var hasObservers = false
+        self.queue.sync {
+            let key = E.key
+            if let observerList = self.observers[key], !observerList.isEmpty {
+                hasObservers = true
+                let postAction = {
+                    self.notificationCenter.post(name: NSNotification.Name(E.key), object: event)
+                }
 
-    /// Registers a listener for a specific event type on a specified scheduler and performs the provided action when that event is emitted.
-    ///
-    /// - Parameters:
-    ///   - eventType: The type of event to listen for.
-    ///   - scheduler: The scheduler on which to perform the action.
-    ///   - action: The action to perform when an event of the specified type is emitted.
-    /// - Returns: A cancellable instance used to unsubscribe from the event.
-    @discardableResult
-    public func onReceive<E: EventRepresentable, S: Scheduler>(_ eventType: E.Type, performOn scheduler: S, action: @escaping (E) -> Void) -> AnyCancellable {
-        subscribeAndNotify(eventType: eventType, scheduler: scheduler, action: action)
+                // Posts the event asynchronously on the provided queue or immediately if no queue is provided.
+                if let queue = queue {
+                    queue.async(execute: postAction)
+                } else {
+                    postAction()
+                }
```

Comment:
Instead of using a `DispatchQueue` directly, could you instead use the SDK's `ThreadUtil` class? I think that could make this function's use less complex.

Reply:
The comment explains the reasoning for this `if/else` well. I don't have a suggestion on that. My suggestion was to see if we could make this function call easier to use by abstracting away the `DispatchQueue` class. 

```suggestion
func post<E>(_ event: E, onMainThread: Bool = false) -> Bool {
        ...
        if onMainThread {
           threadUtil.onMainThread {
             postAction()
           }
        } else {
          postAction() 
        }
```

Reply:
Non-blocker side note: I was expecting that the observer of the event was who would determine what thread to handle a eventbus event on. I don't think the code that posts the event needs to determine what thread it should be handled on?